dashboard/app/models/levels/level.rb

```diff
@@ -855,6 +855,9 @@ def summarize_for_lab2_properties(script, script_level = nil, current_user = nil
     properties_camelized[:usesProjects] = try(:is_project_level) || channel_backed?
     properties_camelized[:finishUrl] = script_level.next_level_or_redirect_path_for_user(current_user) if script_level
     properties_camelized[:baseAssetUrl] = Blockly.base_url
+    properties_camelized[:hasValidation] = validated?
+    properties_camelized[:isAssessment] = script_level&.assessment
+    properties_camelized[:progressionType] = script_level&.primm_progression_type
```

Comment:
`ai_tutor_available` is already on the serialized attributes on the level (see https://github.com/code-dot-org/code-dot-org/blob/b49cc801f82b45fa443ece4c5ca91cb5d1125c19/dashboard/app/models/levels/level.rb#L97) and was already available in the `level_properties`.

It looks like I can pull these off the script_level and the level the same way that we do for AI Tutor (see: https://github.com/code-dot-org/code-dot-org/blob/b49cc801f82b45fa443ece4c5ca91cb5d1125c19/dashboard/app/views/levels/show.html.haml#L19-L20), but I'm not sure the best way to test the others. I'm able to verify hasValidation easily manually, but most of the other PythonLab levels we have locally have "other" as the progression type and aren't assessment levels...

Reply:
Yeah I don't know if any lab2 levels are assessment levels by definition. If the values are always undefined/false/nil that seems plenty safe for now.

Reply:
I hadn't considered that none of the assessment levels are Lab2 levels. That's probably true, and I can double-check with Jamila. When I tested, the values were always `false`.